backend/dataall/modules/shares_base/services/share_notification_service.py

```diff
@@ -109,6 +110,38 @@ def notify_persistent_email_reminder(self, email_id: str):
         )
         return notifications
 
+    def notify_managed_policy_limit_exceeded_action(self, email_id: str):
+        share_link_text = ''
+        if os.environ.get('frontend_domain_url'):
+            share_link_text = (
+                f'<br><br>Please visit data.all <a href="{os.environ.get("frontend_domain_url")}'
+                f'/console/shares/{self.share.shareUri}">share link</a> '
+                f'to view more details.'
+            )
+
+        msg_intro = f"""Dear User,
+        
+        We are contacting you because for a share requested by {email_id} failed because no new managed policy can be attached to your IAM role {self.share.principalRoleName}.
+        Please check the service quota for the managed policies that can be attached to a role in your aws account and increase the limit.
+        For reference please take a look at this link - https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entities
+    
+        Note - Previously made shares are not affected but all new shares will be failed till the time you increase the IAM quota limit.
```

Comment:
nit: all new shares will fail if the IAM Managed Policies quota is not increased or if you do not dettach other IAM policies from the requester IAM role